cocotb/drivers/__init__.py

```diff
@@ -221,26 +220,22 @@ class BusDriver(Driver):
                 bus-signals in an interface or a ``modport``.
                 (untested on ``struct``/``record``, but could work here as well).
             clock (SimHandle): A handle to the clock associated with this bus.
-            array_idx (int or None, optional): Optional index when signal is an array.
-            bus_separator (str, optional): Character(s) to use as separator between bus
-                name and signal name. Defaults to '_'.
+            **kwargs (dict): kwargs is passed on to __init__ of the created .bus field.
+                See ``cocotb.Bus.__init__()`` for more information.
 
     """
     
     _optional_signals = []
 
     def __init__(self, entity, name, clock, **kwargs):
-        # emulate keyword-only arguments in python 2
-        index = kwargs.pop("array_idx", None)
-        bus_separator = kwargs.pop("bus_separator", "_")
-        reject_remaining_kwargs('__init__', kwargs)
+        index = kwargs.get("array_idx")
+        kwargs["optional_signals"] = self._optional_signals + kwargs.get("optional_signals", [])
```

Comment:
It's not clear to me why this makes sense - what is the use case of passing extra optional signals here?